### PR TITLE
[Repo] Add checking caps compatibility between repo_src & repo_sink

### PR DESCRIPTION
**Pull request by jijoongmoon**
# PR Description

It is difficult to check caps compatibility during negotiation because
they are not pipelined each other. That's why reposrc needs caps
property by the way. However, we have to check compatibility somehow during
runtime. In order to do that, GstMetaRepo is introduced.
In this PR, functions and defines are included to manipulate the meta
for repo. During generating buffer in reposink, insert GstMetaRepo
data with GstCaps and push to repo. After that, reposrc extracts the 
GstMetaRepo when pulling the gstbuffer from repo
and compare it with caps defined by user. If they are not compatible, it
shows error and return GST_FLOW_EOS.

**Changes proposed in this PR:**
- Add GstMetaRepo type and manipulation functions
- Add insert GstMetaRepo into buffer
- Add comparison routine during get gstbuffer from repo
- Add Error handling if they are not compatible

Resolves: Issue #1026

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

**Comment by jijoongmoon** [comment]
Test cases is coming up shortly.

File: gst/nnstreamer/tensor_repo.h [hunk 1]
```diff
@@ -32,6 +32,44 @@
 #include <gst/gst.h>
 
 G_BEGIN_DECLS
+/**
+ * @brief GstTensorRepo meta structure.
+ */
+    typedef struct
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_repo.h]: fix indentation

File: gst/nnstreamer/tensor_repo.c [hunk 2]
```diff
@@ -86,17 +175,30 @@ gst_tensor_repo_add_repodata (guint nth)
...
+    GST_TENSOR_REPO_UNLOCK (nth);
+    return FALSE;
+  }
+
   while (data->buffer != NULL) {
     GST_TENSOR_REPO_WAIT_PULL (nth);
   }
 
-  data->buffer = buffer;
+  data->buffer = gst_buffer_copy (buffer);
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_repo.c]: gst_buffer_copy() copies only the data in buffer.
line 195 GstMetaRepo *meta = GST_META_REPO_GET (data->buffer); meta maybe always null.

Comment (jijoongmoon) [reply gst/nnstreamer/tensor_repo.c]: Right. I will delete if(!meta). Thanks.

**Comment by jijoongmoon** [comment]
Ready for further Review!

File: gst/nnstreamer/tensor_repo.c [hunk 3]
```diff
@@ -86,17 +175,28 @@ gst_tensor_repo_add_repodata (guint nth)
...
+
+  GstMetaRepo *meta = GST_META_REPO_GET (data->buffer);
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_repo.c]: GstMetaRepo *meta = GST_META_REPO_ADD (data->buffer);

GST_META_REPO_GET is unnecessary.

Comment (jijoongmoon) [reply gst/nnstreamer/tensor_repo.c]: right. I will remove it. Thanks.

**Pull request merged**